### PR TITLE
update LAPACK easyblock to keep control of compiler options for versions >= 3.9.0

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _LAPACK-3.8.0-GCC-7.3.0-2.30.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2672.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/9dcde55ebe3e1b2543e45a27bea619ae for a full test report.

**Comment by boegel** [comment]
Oh hold on, the `NOOPT` variable was also renamed to `FFLAGS_NOOPT`...

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _LAPACK-3.8.0-GCC-7.3.0-2.30.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2672.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/1e6f3aae6d3225403f4b8a06f49a463e for a full test report.

File: easybuild/easyblocks/l/lapack.py [hunk 1]
```diff
@@ -100,11 +100,32 @@ def configure_step(self):
...
+        self.cfg.update('buildopts', 'CFLAGS="%s -m64"' % os.getenv('CFLAGS'))
+
+        # Fortran compiler flags are controlled via OPTS in older version, via FFLAGS in newer versions
+        if LooseVersion(self.version) >= LooseVersion('3.9.0'):
+            fflags_var = 'FFLAGS'
+            noopt_var = 'FFLAGS_NOOPT'
+        else:
+            fflags_var = 'OPTS'
+            noopt_var = 'NOOPT'
+        self.cfg.update('buildopts', '%s="%s -m64"' % (fflags_var, os.getenv('FFLAGS')))
```
Comment (akesandgren) [inline easybuild/easyblocks/l/lapack.py:123]: Not sure if explicitly setting -m64 is a good thing here and below.

Comment (boegel) [reply easybuild/easyblocks/l/lapack.py:123]: I'm happy to drop this, but it's just copying what the LAPACK `Makefile` does.

Any specific reason you think this is not a good idea?

Comment (akesandgren) [reply easybuild/easyblocks/l/lapack.py:123]: There is no -m64 anywhere in the LAPACK files whatsoever.
-m64 is compiler specific, not all compilers has that flag.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _LAPACK-3.8.0-GCC-7.3.0-2.30.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3140.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/a4b3f6827a79d4fc2ad8d0f4205398af for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**